app/webpacker/controllers/talk-to-us_controller.js

```diff
@@ -46,7 +46,46 @@ export default class extends Controller {
   }
 
   showZendeskChat() {
-    window.$zopim.livechat.window.show();
+    const originalText = this.setLoadingButtonText();
+    this.appendZendeskScript();
+    this.waitForZendesk(() => {
+      window.$zopim.livechat.window.show();
+
+      setTimeout(() => {
+        this.buttonTarget.querySelector('span').innerHTML = originalText;
+      }, 500); // Small delay to account for the chat box animating in.
+    });
+  }
+
+  setLoadingButtonText() {
+    const originalText = this.buttonTarget.querySelector('span').textContent;
+
+    if (!this.zendeskScriptLoaded) {
+      this.buttonTarget.querySelector('span').textContent = 'Starting chat...';
+    }
+
+    return originalText;
+  }
+
+  appendZendeskScript() {
+    if (this.zendeskScriptLoaded) {
+      return;
+    }
+
+    const script = document.createElement('script');
+    script.setAttribute('id', 'ze-snippet');
+    script.src =
+      'https://static.zdassets.com/ekr/snippet.js?key=34a8599c-cfec-4014-99bd-404a91839e37';
+    document.body.appendChild(script);
+  }
+
+  waitForZendesk(callback) {
```

Comment:
There is async/await in JS now if we want to avoid nesting callbacks. Might be neater?
```
 async waitForZendesk() {
    const interval = setInterval(() => {
      if (window.$zopim && window.$zopim.livechat) {
        clearInterval(interval);
      }
    }, 100);
  }

async showZendeskChat() {
  await this.waitForZendesk()
  ...
}
```

Reply:
Actually, it seems `setInterval` doesn't work nicely with Promises/async stuff because it can trigger multiple times so nevermind 🤷